pkg/apis/resource/common.go

```diff
@@ -31,6 +31,7 @@ func DeleteResources(req CollectionDeleteRequest, mc model.ClientSet, kubeConfig
 
 		resources, err := tx.Resources().Query().
```

Comment:
What about we group these resources by their environment and do TopologicalSort per group?